src/components/radio-button/radio-button.component.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback } from "react";
+import { MarginProps } from "styled-system";
+import invariant from "invariant";
+import { CommonCheckableInputProps } from "../../__internal__/checkable-input";
+import RadioButtonStyle from "./radio-button.style";
+import CheckableInput from "../../__internal__/checkable-input/checkable-input.component";
+import RadioButtonSvg from "./radio-button-svg.component";
+import { filterStyledSystemMarginProps } from "../../style/utils";
+import { TooltipProvider } from "../../__internal__/tooltip-provider";
+
+interface InternalRadioButtonProps {
+  inline?: boolean;
+  required?: boolean;
+}
+
+export interface RadioButtonProps
+  extends CommonCheckableInputProps,
+    MarginProps {
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-component"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-element"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-role"?: string;
+  /** When true, sets the component in line (for RadioButtonGroup) */
+  inline?: boolean;
```

Comment:
`inline` is already defined in `InternalRadioButtonProps`, also I'm not sure what is the reason for either `inline` or `required` being internal props? They can actually be used by library consumers can't they?

Reply:
this prop should be passed down from the `RadioButtonGroup` component, the same goes for the 'required` prop. Looks like it was left there as a mistake.